stats/opencensus/e2e_test.go

```diff
@@ -0,0 +1,773 @@
+					{
+						Tags: []tag.Tag{
+							{
+								Key:   smtk,
+								Value: "grpc.testing.TestService/UnaryCall",
+							},
+						},
+						Data: &view.DistributionData{
+							Count:          1,
+							CountPerBucket: []int64{0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0},
+						},
+					},
+					{
+						Tags: []tag.Tag{
+							{
+								Key:   smtk,
+								Value: "grpc.testing.TestService/FullDuplexCall",
+							},
+						},
+						Data: &view.DistributionData{
+							Count:          1,
+							CountPerBucket: []int64{1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0},
+						},
+					},
+				},
+			},
+		},
+
+		// TODO: 3 missing latency metrics, figure out a way to test even though
+		// non deterministic, perhaps ignore exact distribution and test
+		// everything else?
+
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			view.Register(test.metric)
+			// To cleanup global state between testing iterations, to prevent
+			// measures persisting over multiple testing iterations.
+			defer view.Unregister(test.metric)
+			mri := time.Millisecond * 100
+			view.SetReportingPeriod(mri)
+			fe := &fakeExporter{
+				t:         t,
+				seenViews: make(map[string]*viewInformation),
+			}
+			view.RegisterExporter(fe)
+			defer view.UnregisterExporter(fe)
+
+			ss := &stubserver.StubServer{
+				UnaryCallF: func(ctx context.Context, in *grpc_testing.SimpleRequest) (*grpc_testing.SimpleResponse, error) {
+					return &grpc_testing.SimpleResponse{}, nil
+				},
+				FullDuplexCallF: func(stream grpc_testing.TestService_FullDuplexCallServer) error {
+					for {
+						_, err := stream.Recv()
+						if err == io.EOF {
+							return nil
+						}
+					}
+				},
+			}
+			if err := ss.Start([]grpc.ServerOption{ServerOption(TraceOptions{})}, DialOption(TraceOptions{})); err != nil {
+				t.Fatalf("Error starting endpoint server: %v", err)
+			}
+			defer ss.Stop()
+			ctx, cancel := context.WithTimeout(context.Background(), defaultTestTimeout)
+			defer cancel()
+			// Make two RPC's, a unary RPC and a streaming RPC. These should cause
+			// certain metrics to be emitted.
+			if _, err := ss.Client.UnaryCall(ctx, &grpc_testing.SimpleRequest{Payload: &grpc_testing.Payload{}}); err != nil {
+				t.Fatalf("Unexpected error from UnaryCall: %v", err)
+			}
+			stream, err := ss.Client.FullDuplexCall(ctx)
+			if err != nil {
+				t.Fatalf("ss.Client.FullDuplexCall failed: %f", err)
+			}
+
+			stream.CloseSend()
+			if _, err = stream.Recv(); err != io.EOF {
+				t.Fatalf("unexpected error: %v, expected an EOF error", err)
+			}
+			if errs := seenViewMatches(fe, test.metric.Name, test.wantVI); len(errs) != 0 {
+				t.Fatalf("Invalid OpenCensus export data: %v", errs)
+			}
+		})
+	}
+}
+
+// unit test for unmarshal JSON using map
+
+// should I test opencensus progpagted over wire
+
+// trace context plumbed as an attribute is irrelevant because I don't even deal with traces at this moment.
```

Comment:
Will delete. Won't push commit in case middle of review.

Reply:
New commits are fine..just not force pushes, which shouldn't ever be necessary anyway.